src/gpu/generic/sycl/ref_sum.hpp

```diff
@@ -64,9 +64,9 @@ struct ref_sum_t : public gpu::generic::sycl::primitive_t {
                                 && src_d.ndims() <= xpu::sycl::md_t::max_dims,
                         VERBOSE_UNSUPPORTED_TENSOR_LAYOUT, "src");
 
-                VDISPATCH_SUM(attr()->scales_.has_default_values()
-                                || is_supported_type(
-                                        scales.get_data_type(DNNL_ARG_SRC + i)),
+                VDISPATCH_SUM(IMPLICATION(!attr()->scales_.has_default_values(),
+                                      is_supported_type(scales.get_data_type(
+                                              DNNL_ARG_SRC + i))),
```

Comment:
(nit) Isn't that the same since `IMPLICATION(a, b)` is equivalent to `!a || b` ?

Reply:
I can't remember why I added this change, I think it was mainly for readability, but yes, you are right that it is the same check. I could revert it if preferable?

Reply:
This was just to make sure I was not missing something thanks.